### ISSUE TITLE
Hidden cursor stays hidden on macOS when neovide loses focus

### ISSUE THREAD

**Issue by jmaliksi** [neovide/neovide#2414]
**Describe the bug**
After turning on `neovide_hide_mouse_when_typing`, the mouse cursor sometimes remains hidden when moving focus between other desktop apps and neovide. 

**To Reproduce**
1. Focus on neovide window
2. alt tab to move focus to a different window
3. move mouse over unfocused neovide window
4. alt tab or click to bring focus back to neovide. Cursor disappears (bug)
5. doing a hjhjhjhj and moving the mouse unsticks the mouse hidden state and cursors appears again (workaround)

**Expected behavior**
Mouse cursor should appear as soon as any mouse event occurs

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS 14.4
 - Neovide Version: 0.12.2—compiled from `main` this morning
 - Neovim Version: v0.9.4

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**
(not actually sure how helpful the log will be and it also seems to be full of plugin noise. see additional context for more info, but also happy to provide the log if this might be more of a problem with my setup than the winit bug)

**Additional context**
It looks like the root cause is https://github.com/rust-windowing/winit/issues/2539 though [this comment](https://github.com/neovide/neovide/issues/1966#issuecomment-1686033380) mentioned that it had only been observed in macOS 13. Unfortuntely I have triggered it in macOS 14.4 so the bug seems to be alive and well.

I was able to make the mouse behave how I wanted by checking for window focus before hiding the cursor, and so was able to avoid `self.mouse_hidden` getting desynced from the "true" window cursor state
```
index fd0f586..9e62024 100644
--- a/src/window/mouse_manager.rs
+++ b/src/window/mouse_manager.rs
@@ -439,7 +439,7 @@ impl MouseManager {
                     renderer,
                     window,
                 );
-                if self.mouse_hidden {
+                if self.mouse_hidden && window.has_focus() {
                     window.set_cursor_visible(true);
                     self.mouse_hidden = false;
                 }
```

I'm super super new to rust and pretty much never do desktop application dev, so I don't know if this is a viable fix or not. Ideally I'd get the true cursor state from the window but I didn't see that in the winit API?

Another thought is on the lose focus window event, always set the mouse to visible since you'll likely not mind (and maybe even prefer) seeing the mouse cursor while flipping through your desktop windows. Though I think that may also be vulnerable to the root problem of `set_cursor_visible` not working properly when the window is unfocused?

**Comment by uloco** [comment]
This is really annoying when opening another application with the launcher too (on macos). So I am in neovide, then got to another app via launcher and when I click neovide again, the mouse disappears.

**Comment by jmaliksi** [comment]
I forgot entirely about this issue as I'd been running the "fix" above, but I just got a new macbook and installed the app (rather than compiling) and looks like the issue is still around. I don't think I observed any adverse effects while using my patch over the past year, so @fredizzimo any hesitations with me opening up a PR with it?

**Comment by fredizzimo** [comment]
Yes, you can send a PR, but I think it would be good to make the same check when hiding the cursor as well.

I would prefer if we didn't need to track this ourselves though, but this issue hasn't been picked up https://github.com/rust-windowing/winit/issues/1295

**Comment by Geobert** [comment]
That was it, I tried to understand the root cause before creating an issue but failed so far x) It was driving me crazy x)

On Windows as well (regarding the labels attached)

**Comment by Geobert** [comment]
Found an edge case: 
- type to hide mouse cursor
- alt-tab
- move mouse to neovide, it is hidden


(I have fix coming in)

**Comment by Geobert** [comment]
I’ll test it tomorrow :) but note that in my case, the window is still not focused, making the mouse disappear :)

**Comment by falcucci** [comment]
@Geobert great let me know, if it persists for you I open again the issue

**Comment by Geobert** [comment]
> [@Geobert](https://github.com/Geobert) great let me know, if it persists for you I open again the issue

Seems good!
